src/libraries/System.Security.Cryptography/tests/X509Certificates/PfxIterationCountTests.cs

```diff
@@ -0,0 +1,227 @@
+                        VerifyThrowsCryptoExButDoesNotThrowPfxWithoutPassword(() => Import(fileName));
+                        VerifyThrowsCryptoExButDoesNotThrowPfxWithoutPassword(() => Import(fileName, password));
+                        VerifyThrowsCryptoExButDoesNotThrowPfxWithoutPassword(() => Import(fileName, PfxTests.GetSecureString(password)));
+                    }
+                }
+            }
+        }
+
+        internal static void VerifyThrowsCryptoExButDoesNotThrowPfxWithoutPassword(Action action)
+        {
+            CryptographicException ce = Assert.ThrowsAny<CryptographicException>(action);
+            Assert.DoesNotContain("2233907", ce.Message);
+        }
+
+        [ConditionalTheory]
+        [MemberData(nameof(GetCertsWith_NonNullOrEmptyPassword_MemberData))]
+        public void Import_NonNullOrEmptyPasswordExpected_Throws(string name, string password, bool usesPbes2, byte[] blob, long iterationCount)
+        {
+            if (usesPbes2 && !PfxTests.Pkcs12PBES2Supported)
+            {
+                throw new SkipTestException(name + " uses PBES2 which is not supported on this version.");
+            }
+
+            CryptographicException ce = Assert.ThrowsAny<CryptographicException>(() => Import(blob));
+
+            if (PfxTests.IsPkcs12IterationCountAllowed(iterationCount, PfxTests.DefaultIterations))
+            {
+                Assert.NotNull(Import(blob, password));
+                Assert.NotNull(Import(blob, PfxTests.GetSecureString(password)));
+
+
+                using (TempFileHolder tempFile = new TempFileHolder(blob))
+                {
+                    string fileName = tempFile.FilePath;
+                    Assert.NotNull(Import(fileName, password));
+                    Assert.NotNull(Import(fileName, PfxTests.GetSecureString(password)));
+                }
+            }
+        }
+
+        [Fact]
+        public void ExportedPfxWithNullPassword_DecryptReturnsValidPaddingWithEmptyPassword()
+        {
+            Assert.NotNull(Import(TestData.MsCertificateExportedToPfx_NullPassword));
+        }
+
+        [Fact]
+        public void Import_BlobHasMoreThanOnePfx_LoadsOnlyOne()
+        {
+            // These certs don't use PBES2 so they should be supported everywhere.
+            byte[] firstPfx = TestData.Pkcs12WindowsDotnetExportEmptyPassword.HexToByteArray();
+            Assert.Equal("CN=test", Import(firstPfx).Subject);
+
+            byte[] secondPfx = TestData.Pkcs12Builder3DESCBCWithEmptyPassword.HexToByteArray();
+            Assert.Equal("CN=potato", Import(secondPfx).Subject);
+
+            byte[] twoPfxes = new byte[firstPfx.Length + secondPfx.Length];
+            Array.Copy(firstPfx, twoPfxes, firstPfx.Length);
+            Array.Copy(secondPfx, 0, twoPfxes, firstPfx.Length, secondPfx.Length);
+
+            Assert.Equal("CN=test", Import(twoPfxes).Subject);
+        }
+
+        internal static readonly List<PfxInfo> s_Certificates = GetCertificates();
```

Comment:
```suggestion
        internal static readonly List<PfxInfo> s_certificates = GetCertificates();
```

Does it need to be `internal`, or could it be `private`/`protected`?

Reply:
It needs to be internal for `s_certificatesDictionary` in `PfxIterationCountTests_CustomAppDomainDataLimit`.